networking_dvs/utils/dvs_util.py

```diff
@@ -508,8 +511,42 @@ def update_dvportgroup(self, pg_ref, config_version, port_config=None, name=None
                 config_version = vim_util.get_object_property(
                     self.connection.vim, pg_ref, "config.configVersion")
                 return self.update_dvportgroup(pg_ref, config_version, port_config, name, retries-1)
+            if dvs_const.BULK_FAULT_TEXT in str(e):
+                info = get_task_info(self.connection, pg_update_task)
+                self.rectifyForFault(info.error.fault)
+                return
             raise exceptions.wrap_wmvare_vim_exception(e)
 
+    def rectifyForFault(self, fault):
+        """
+        Handles DvsOperationBulkFault by attempting to rectify the hosts' configuration with the switch
+        """
+        host_faults = getattr(fault, "hostFault", None)
+        hosts = set()
+        for hf in host_faults:
+            if not hf:
+                continue
+            host_ref = hf.host.value
+            if host_ref in self.hosts_to_rectify:
+                if time.time() - self.rectify_wait > self.hosts_to_rectify[host_ref]:
+                    self.hosts_to_rectify[host_ref] = time.time()
+                    hosts.add(host_ref)
+                else:
+                    LOG.debug("Timeout for host {} is not reached yet, skipping.".format(host_ref))
+            else:
+                self.hosts_to_rectify[host_ref] = time.time()
+                hosts.add(host_ref)
+
+        if not hosts:
+            return
+        LOG.debug("Hosts to rectify: {}".format(hosts))
+        rectify_task = self.connection.invoke_api(
+                    self.connection.vim,
+                    "RectifyDvsOnHost_Task",
+                    self._service_content.dvSwitchManager,
+                    hosts=list(hosts))
+        self.connection.wait_for_task(rectify_task)
```

Comment:
Correct, there's no need to wait for it here. Downside is that we won't log task failures in the agent, though they would still be present in the vcenter's log.